### ISSUE TITLE
Room not backfilling?

### ISSUE THREAD

**Issue by z3ntu** [matrix-org/dendrite#1565]
### Background information

- **Dendrite version or git SHA**: 0.2.1
- **Monolith or Polylith?**: Monolith
- **SQLite3 or Postgres?**: SQLite3
- **Running in Docker?**: No
- **`go version`**: 1.15.3

### Description

I've joined a room (invited my dendrite user into a room) but I don't seen any messages before the moment I've joined in the room even though I believe I should be able to read the history 
![image](https://user-images.githubusercontent.com/3768500/97108507-58927780-16ce-11eb-98e2-f18eff39c336.png)

I see two errors in my dendrite log; they look related:
```
time="2020-10-25T13:16:56.412565185Z" level=error msg="Error checking if allowed to see event" func=github.com/matrix-org/dendrite/roomserver/internal/helpers.ScanEventTree file="github.com/matrix-org/dendrite/roomserver/internal/helpers/helpers.go:309" error="LoadStateAtEvent.SnapshotNIDFromEventID($vdu0k0VrOWhkdaxWO3SqCyiSArGeKTpKKhaXyKuveOM) returned 0 NID, was this event stored?" event_id="$vdu0k0VrOWhkdaxWO3SqCyiSArGeKTpKKhaXyKuveOM" req.id=kvm2EzHDGsel req.method=POST req.path="/_matrix/federation/v1/get_missing_events/!VYsYJQNEpQSJlMMxiC:fam-ribbers.com" server=librem.one
time="2020-10-25T13:17:40.426637713Z" level=error msg="Error checking if allowed to see event" func=github.com/matrix-org/dendrite/roomserver/internal/helpers.ScanEventTree file="github.com/matrix-org/dendrite/roomserver/internal/helpers/helpers.go:309" error="LoadStateAtEvent.SnapshotNIDFromEventID($vdu0k0VrOWhkdaxWO3SqCyiSArGeKTpKKhaXyKuveOM) returned 0 NID, was this event stored?" event_id="$vdu0k0VrOWhkdaxWO3SqCyiSArGeKTpKKhaXyKuveOM" req.id=KhG33Rn0SN62 req.method=POST req.path="/_matrix/federation/v1/get_missing_events/!VYsYJQNEpQSJlMMxiC:fam-ribbers.com" server=librem.one
```

### Steps to reproduce

- join #offtopic:postmarketos.org ?

**Comment by z3ntu** [comment]
This looks like interesting logs:
```
time="2020-10-25T13:44:40.758543498Z" level=info msg="ServersAtEvent history visibility not shared: world_readable" func=github.com/matrix-org/dendrite/roomserver/internal/perform.joinEventsFromHistoryVisibility file="github.com/matrix-org/dendrite/roomserver/internal/perform/perform_backfill.go:530"
time="2020-10-25T13:44:40.759023062Z" level=info msg="ServersAtEvent including 0 current events from history visibility" func="github.com/matrix-org/dendrite/roomserver/internal/perform.(*backfillRequester).ServersAtEvent" file="github.com/matrix-org/dendrite/roomserver/internal/perform/perform_backfill.go:434"
```

https://github.com/matrix-org/dendrite/blob/0f7e707f399e7f633c58f4e1a5aedc0e45f90241/roomserver/internal/perform/perform_backfill.go#L530

**Comment by apexo** [comment]
History backfilling was implemented in 4ad52c67cacc21997f49decd57e3105beb8ab62d for "shared" visibility. Don't know why world_readable wasn't considered, probably just an oversight.
